### PR TITLE
Expose format string functions directly to users (addressing issue #32)

### PR DESCRIPTION
**Pull request by jackirvine97**
This pull request:

- Creates a new module, `format_str_case.py`, which collects the relevant functions that were previously in `utils.py`
- Exposes format string function directly to users:

`>>> from pyproprop import format_str_case`

- Refactors string formatting test cases into parameterisable fixtures (using `pytest-cases`)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ test_script:
   - python setup.py install
 
   # Run the test suite
-  - conda install --yes --file=tests\requirements.txt
+  - FOR /F "delims=~" %%F in (tests\requirements.txt) DO (conda install -q -y %%F || pip install %%F)
```
Comment (brocksam) [inline appveyor.yml:73]: Related to _pytest-cases_.

File: .travis.yml [hunk 2]
```diff
@@ -23,9 +23,8 @@ install:
   - conda create -q -n install-env python=$TRAVIS_PYTHON_VERSION
   - source activate install-env
   # Taken from https://stackoverflow.com/questions/35802939/install-only-available-packages-using-conda-install-yes-file-requirements-t
-  # - while read requirement; do conda install -q -y $requirement || pip install $requirement; done < requirements.txt
   - conda install -q -y --file=requirements.txt
-  - conda install -q -y --file=tests/requirements.txt
+  - while read requirement; do conda install --yes $requirement || pip install $requirement; done < tests/requirements.txt
```
Comment (brocksam) [inline .travis.yml:27]: Related to _pytest-cases_.

File: tests/conftest.py [hunk 3]
```diff
@@ -2,9 +2,57 @@
 import sys
 
 import pytest
+from pyproprop import processed_property
```
Comment (brocksam) [inline tests/conftest.py:5]: Not sure this should be here, think it goes separated below as it's first party not third party... I could be wrong though.

**Review by brocksam — COMMENTED** [review]
Inclusion of _pytest-cases_ needs to be discussed.

**Comment by jackirvine97** [comment]
Pytest-cases conda-forge issue: https://github.com/smarie/python-pytest-cases/issues/135

File: tests/conftest.py (hunk 3 above)
Comment (jackirvine97) [reply tests/conftest.py]: Refactored in 3672345

**Review by brocksam — APPROVED** [review]

**Pull request merged**